### ISSUE TITLE
Checkov unable to resolve CFN Parameters and Conditions

### ISSUE THREAD

**Issue by cbhat-ie** [bridgecrewio/checkov#5325]
**Describe the issue**
Framework: Cloudformation
Checkov is unable to resolve the CFN parameters and conditions properly like in the example shown below. Although 'public-access-block' evaluates to true, the scan still complains about "Ensure S3 bucket has block public ACLS enabled". Any idea why?

**Example Template**
```yaml
  BlockPublicAccess:
    Description: Set to true to block all public access with a public access block configuration
    Type: String
    AllowedValues:
      - "true"
      - "false"
    Default: "true"

  public-access-block: !Equals
    - !Ref BlockPublicAccess
    - "true"

  S3Bucket:
    DeletionPolicy: Delete
    UpdateReplacePolicy: Delete
    Type: AWS::S3::Bucket
      PublicAccessBlockConfiguration: !If
        - public-access-block
        - BlockPublicAcls: true
          BlockPublicPolicy: true
          IgnorePublicAcls: true
          RestrictPublicBuckets: true
        - !Ref "AWS::NoValue"
```

**Comment by cbhat-ie** [comment]
Hi, any idea why "public-access-block" is not being evaluated?

**Comment by zsaltys** [comment]
This is definitely still an issue and should be reopened.

**Comment by emondal1026** [comment]
agreed this is still an issue. I will try to dig in more to provide some better context. But there have been many false positives reported by the checkov linter due to inability to resolve references or interpret function key words (such as !Or operator or !Equals).

**Comment by emondal1026** [comment]
it is also seemingly inconsistent sometimes? For example, this got caught by a check:
`  DBSNSTopic:
    Type: AWS::SNS::Topic
    Properties:
      KmsMasterKeyId: !Ref KmsKeyId
      Subscription:
        - Endpoint: !Ref 'SnsToSlackLambdaArn'
          Protocol: lambda
      Tags:
        -
          Key: EnvLabel
          Value: !Ref EnvLabel`

But this passed:

`    BeanstalkWebApiSNSTopic:
      Condition: EnableHealthNotifications
      Type: AWS::SNS::Topic
      Properties:
        TopicName: !Sub '${EnvLabel}-BeanstalkWebApi'
        KmsMasterKeyId: !Ref KmsKeyId
        Subscription:
          - Endpoint: !Ref 'SnsToSlackLambdaArn'
            Protocol: lambda`

 CKV_AWS_26: "Ensure all data stored in the SNS topic is encrypted"

**Comment by zsaltys** [comment]
I think what happens is that when there's a condition it simply "ignores" the file and catches nothing in it..

**Comment by cbhat-ie** [comment]
There appears to be a significant limitation in Checkov regarding the scanning of files with conditional statements in them. Given that using conditions is a common practice in CloudFormation files, what recommendation does Checkov provide for addressing this issue? @emondal1026 , @gruebel

**Comment by cbhat-ie** [comment]
I would like this issue to be re-opened again please... this is a significant gap in the tool. Currently if a condition or an intrinsic CFN function like `!If` is present in the code, then the Checkov scan quietly skips the file. In comparison, tools like cfn-nag and sonarqube are able to scan the files and report issues.

If not able to fix this gap, Checkov scans at the least should report that a CFN framework file was skipped and not scanned so the developers using the tool are aware and do not think that the file was scanned and no issues were reported.
@emondal1026 , @gruebel
